zk-keygen/src/main.rs

```diff
@@ -70,6 +75,85 @@ fn app(crate_version: &str) -> Command {
                 .key_generation_common_args()
                 .arg(no_outfile_arg().conflicts_with_all(&["outfile", "silent"]))
         )
+        .subcommand(
+            Command::new("pubkey")
+                .about("Display the pubkey from a keypair file")
+                .disable_version_flag(true)
+                .arg(
+                    Arg::new("keypair")
+                        .index(1)
+                        .value_name("KEYPAIR")
+                        .takes_value(true)
+                        .help("Filepath or URL to a keypair"),
+                )
+                .arg(
+                    Arg::new(SKIP_SEED_PHRASE_VALIDATION_ARG.name)
+                        .long(SKIP_SEED_PHRASE_VALIDATION_ARG.long)
+                        .help(SKIP_SEED_PHRASE_VALIDATION_ARG.help),
+                )
+                .arg(
+                    Arg::new("type")
+                        .long("type")
+                        .takes_value(true)
+                        .possible_values(["elgamal"])
+                        .value_name("TYPE")
+                        .required(true)
+                        .help("The type of keypair")
+                )
+                .arg(
+                    Arg::new("outfile")
+                        .short('o')
+                        .long("outfile")
+                        .value_name("FILEPATH")
+                        .takes_value(true)
+                        .help("Path to generated file"),
+                )
+                .arg(
+                    Arg::new("force")
+                        .long("force")
+                        .help("Overwrite the output file if it exists"),
+                )
```

Comment:
this might be something from `solana-keygen pubkey` better left there. i've never used (forgot it existed, in fact) output to file 'cause anytime i want to write out a pubkey, i want it in base58 and i'm just going to `> outfile`.

i could see the value in adding (if deemed needed in the future(?)) an `--output json` to get to json, since that's nontrivial to conversion in shell

Reply:
Ah yes, I think that is a good point. Let's just remove the `--outfile` flag for the `pubkey` command.